connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -856,6 +858,47 @@ public void run() {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testSinkTasksHandleCloseErrors() throws Exception {
```

Comment:
Can you also write a test where an exception is thrown only by `sinkTask.close`? Actually, we could keep the name for this new test here _as is_, and the new test could be named in a way that tells us that the exceptions on close are suppressed in the presence of exceptions in the main try block.

Reply:
Ive written the test as-is, and the exception from close is caught by WorkerTask::doRun.

However, it never gets wrapped in a ConnectException like exceptions from the other connector methods, but i'm not sure that it's in-scope to change this in this PR.

Reply:
Let me see if I understand what you are describing as `wrapped`. 
My use case is as follows: 
`SinkTask#close` attempts to release resources and if it fails it throws a `ConnectException` as we'd expect from connector developers to do (currently it throws a `RuntimeException` which might be less representative). 

With your fix this exception can appear as suppressed when an exception happens in `SinkTask#put` and that's what your test is guarding against. 

My point is to add the missing test case for when the exception on `close` is the only exception that is thrown. There is a variety of ways to do that, but I agree with you, this test is not there now. However, I don't think this necessarily makes it out of scope.

Reply:
I added that test, so please look at the diff.

By wrapping, i'm completely disregarding the behavior from the connector, I'm only discussing the additional layers of exceptions added by the framework before printing.

At the moment, the RuntimeException from put is wrapped by a ConnectException, but the RuntimeException from close is never wrapped in a ConnectException. I'm questioning whether this is the ideal behavior, and whether we should add that wrapping layer, or consider it out-of-scope for this PR.

Reply:
That's the testing I had in mind. That's great. 
No need to change the exception type right now.